### PR TITLE
Tweaked search panel design

### LINKED ISSUE #3345: Make search options clearer
> Wouldn't that create the same confusion you have about `Search in string identifiers`?

Why? If `Search in source strings` is disabled it doesn't imply `Search in string identifiers` can't be enabled. Which is the problem

> UPDATE: As for the checkboxes, since we have other 2 "Match…" options, I'm not a fan of the idea.

I meant something like this:

Posnetek zaslona 2024-09-17 ob 09 47 01

_Originally posted by @mathjazz in https://github.com/mozilla/pontoon/issues/3344#issuecomment-2354818812_

### PR DESCRIPTION
**Pull request by harmitgoswami**
Fix #3345 

As per the discussion in the related issue, this PR renames the following in `translate/src/modules/search/constants.ts`:

- `search_translations_only` --> `search_exclude_source_strings`
- `Search in translations only` --> `Exclude source strings`
- `Search in string identifiers` --> `Include string identifiers`
- `Search in rejected translations` --> `Include rejected translations`

Renaming the variables was done to improve clarity and more accurately reflect how the query changes in the backend. Additionally, I've added visual separation between the 'Match' and 'Include/Exclude' filters, reflecting the difference in their behaviours.

image

### CODE REVIEW

**Comment by mathjazz** [comment]
Deploying this and #3349 to stage...

File: pontoon/base/models/entity.py [hunk 1]
```diff
@@ -889,8 +889,8 @@ def for_project_locale(
             # Search in source strings
             entity_filters = (
                 (
-                    Q(pk=None)  # Ensures that no source strings are returned
-                    if search_translations_only
+                    Q(pk__in=[])  # Ensures that no source strings are returned
```
Comment (mathjazz) [inline pontoon/base/models/entity.py:892]: Why this change?

File: translate/src/modules/search/components/SearchPanel.tsx [hunk 2]
```diff
@@ -31,9 +31,18 @@ const SearchOption = ({
   selected: boolean;
   onToggle: () => void;
 }) => {
+  // Find the first option that doesn't have "Match" in the name, and apply top margin for visual seperation
```
Comment (flodolo) [inline translate/src/modules/search/components/SearchPanel.tsx:34]: At risk of asking a dumb question, this feels brittle. 

Can we have IDs associated to these checkboxes, and add a margin to that element?

Comment (harmitgoswami) [reply translate/src/modules/search/components/SearchPanel.tsx:34]: So for example, `Include string identifiers` would get ID = 3, and then we apply the top margin to the element whose ID = 3? Or did I understand incorrectly

**Review by flodolo — APPROVED** [review]
Behavior looks correct to me on stage, only question about how to add the margin (assuming we even want one?)

File: translate/src/modules/search/components/SearchPanel.tsx [hunk 3]
```diff
@@ -31,9 +31,18 @@ const SearchOption = ({
...
+  const separator = SEARCH_OPTIONS.find(
+    (option) => !option.name.includes('Match'),
+  )?.name;
+
```
Comment (mathjazz) [inline translate/src/modules/search/components/SearchPanel.tsx:38]: That looks complex and fragile. I'd just use slug as the className and add margin to `.search_identifiers`.

Alternatively you could have two lists, once for Matches and one for Incudes/Excludes.

File: translate/src/modules/search/components/SearchPanel.tsx (hunk 2 above)
Comment (flodolo) [reply translate/src/modules/search/components/SearchPanel.tsx:34]: Not necessarily a numeric ID, a CSS ID (e.g. `id="option-include-ids"`). Let's wait for @mathjazz though, as I'm not sure if the space is wanted (it made sense when there was text between the two group of options).

File: translate/src/modules/search/components/SearchPanel.tsx (hunk 3 above)
Comment (mathjazz) [reply translate/src/modules/search/components/SearchPanel.tsx:38]: Deleting due to https://github.com/mozilla/pontoon/pull/3352#pullrequestreview-2313063497.

File: translate/src/modules/search/components/SearchPanel.tsx (hunk 2 above)
Comment (mathjazz) [reply translate/src/modules/search/components/SearchPanel.tsx:34]: My deleted (didn't notice this thread earlier) comment:

"That looks complex and fragile. I'd just use slug as the className and add margin to .search_identifiers.

Alternatively you could have two lists, once for Matches and one for Incudes/Excludes."

File: pontoon/base/models/entity.py (hunk 1 above)
Comment (harmitgoswami) [reply pontoon/base/models/entity.py:892]: I read over the [StackOverflow post](https://stackoverflow.com/questions/29900386/how-to-construct-django-q-object-matching-none) again, it seems that using `pk__in=[]` is a safer and potentially less costly alternative than `pk=None`

Comment (mathjazz) [reply pontoon/base/models/entity.py:892]: Nice work with the research!

File: translate/src/modules/search/components/SearchPanel.tsx (hunk 2 above)
Comment (mathjazz) [reply translate/src/modules/search/components/SearchPanel.tsx]: Much clearer now, thanks!

Yeah, it's OK if some classes are unused.

**Comment by mathjazz** [comment]
Could you please also apply these CSS changes copied from Firefox Devtools? Commented out rules need to be removed.

This will make padding and margins consistent with the Filters panel, otherwise switching between the two makes text jump a little.

```
.search-panel {
  & .menu {
    & .title {
      margin: 5px 0;
    }
    & ul {
      /* margin: 10px 0; */
      /* padding: 4px 0; */
      margin: 0;
    }
  }
  & .search-button {
    margin-bottom: 5px;
    margin-top: 8px;
  }
}

&:hover, & .applied-count {
  margin-top: 35px;
}

```

**Review by mathjazz — APPROVED** [review]
